src/main/java/com/microsoft/azure/springcloudplayground/generator/ProjectGenerator.java

```diff
@@ -135,15 +136,29 @@ public File generateProjectStructure(ProjectRequest request) {
         }
     }
 
+    private void servicesPortsUpdate(@NonNull ProjectRequest request) {
+        Assert.isNull(request.getParent(), "should be parent project.");
+
+        final Map<String, Integer> servicesPortsMap = request.getServicesPortsMap();
+        final List<String> services = request.getServices();
+
+        final long configuredCount = services.stream().filter(s -> servicesPortsMap.get(s) != null).count();
+
+        if (configuredCount < services.size()) {
```

Comment:
as discussed, validation happens on client end.

Reply:
will remove this after client end ready.